hash_test.go

```diff
@@ -78,6 +78,59 @@ func TestHash(t *testing.T) {
 			if bytes.Equal(sum, initSum) {
 				t.Error("Write didn't change internal hash state")
 			}
+
+      if _, ok := h.(encoding.BinaryMarshaler); ok {
+				state, err := h.(encoding.BinaryMarshaler).MarshalBinary()
+				if err != nil {
+					t.Errorf("could not marshal: %v", err)
+				}
+				h2 := cryptoToHash(ch)()
+				if err := h2.(encoding.BinaryUnmarshaler).UnmarshalBinary(state); err != nil {
+					t.Errorf("could not unmarshal: %v", err)
+				}
+				if actual, actual2 := h.Sum(nil), h2.Sum(nil); !bytes.Equal(actual, actual2) {
+					t.Errorf("0x%x != marshaled 0x%x", actual, actual2)
+				}
+			}
+
+			if _, ok := h.(interface {
+				AppendBinary(b []byte) ([]byte, error)
+			}); ok {
+				// Create a slice with 10 elements
+				prebuiltSlice := make([]byte, 10)
+				// Fill the slice with some data
+				for i := range prebuiltSlice {
+					prebuiltSlice[i] = byte(i)
+				}
+
+				// Clone the prebuilt slice for comparison
+				prebuiltSliceClone := append([]byte(nil), prebuiltSlice...)
+
+				// Append binary data to the prebuilt slice
+				state, err := h.(interface {
+					AppendBinary(b []byte) ([]byte, error)
```

Comment:
You can avoid this type cast by using a named variable in here https://github.com/golang-fips/openssl/pull/170/files#diff-848945eacf9cfc5d080eda9e5c42ac8adc3afbe413745df09dbae08ae011a47aR97